### PR TITLE
Introduce Get/Set Strategy for Personalization

### PR DESCRIPTION
**Pull request by julienbourdeau**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | yes    
| BC breaks?        | no     
| Related Issue     | Fix #...  
| Need Doc update   | yes

## Describe your change

Introducing

```php
$strategy = $searchClient->getStrategy();
$searchClient->setStrategy($strategy);
```

### CODE REVIEW

File: src/SearchClient.php [hunk 1]
```diff
@@ -242,6 +242,23 @@ public static function generateSecuredApiKey($parentApiKey, $restrictions)
...
+    public function setStrategy($strategy, $requestOptions = array())
+    {
+        $apiResponse = $this->api->write(
+            'POST',
+            api_path('1/recommendation/personalization/strategy'),
+            $strategy,
+            $requestOptions
+        );
+
+        return $apiResponse; // TODO: Create WaitableResponse once we know
```
Comment (nunomaduro) [inline src/SearchClient.php]: Yeah, as you specified missing response object here.

File: src/SearchClient.php [hunk 2]
```diff
@@ -242,6 +242,23 @@ public static function generateSecuredApiKey($parentApiKey, $restrictions)
         return base64_encode($content);
     }
 
+    public function getStrategy($requestOptions = array())
+    {
+        return $this->api->read('GET', api_path('/1/recommendation/personalization/strategy'), $requestOptions);
+    }
+
+    public function setStrategy($strategy, $requestOptions = array())
```
Comment (nunomaduro) [inline src/SearchClient.php]: Missing test for this.

File: src/SearchClient.php [hunk 3]
```diff
@@ -242,6 +242,23 @@ public static function generateSecuredApiKey($parentApiKey, $restrictions)
         return base64_encode($content);
     }
 
+    public function getStrategy($requestOptions = array())
```
Comment (nunomaduro) [inline src/SearchClient.php]: Missing test for this.

File: src/SearchClient.php [hunk 4]
```diff
@@ -242,6 +242,23 @@ public static function generateSecuredApiKey($parentApiKey, $restrictions)
...
+    {
+        return $this->api->read('GET', api_path('/1/recommendation/personalization/strategy'), $requestOptions);
+    }
+
+    public function setStrategy($strategy, $requestOptions = array())
+    {
+        $apiResponse = $this->api->write(
+            'POST',
+            api_path('1/recommendation/personalization/strategy'),
+            $strategy,
```
Comment (tkrugg) [inline src/SearchClient.php]: The name of the function `getStrategy`/`setStrategy` suggests that they are generic.
Therefore shouldn't `personalization` be a parameter?

Comment (julienbourdeau) [reply src/SearchClient.php]: So you'd rather see `setPersonalizationStrategy`?

Comment (tkrugg) [reply src/SearchClient.php]: I think so, but I don't know if there is going to be any more use cases for strategies

File: src/SearchClient.php (hunk 3 above)
Comment (nunomaduro) [inline src/SearchClient.php]: This should be, as discussed together: `getPersonalizationStrategy`.

File: src/SearchClient.php (hunk 2 above)
Comment (nunomaduro) [inline src/SearchClient.php]: This should be, as discussed together: `setPersonalizationStrategy`.

File: src/SearchClient.php (hunk 1 above)
Comment (nunomaduro) [reply src/SearchClient.php]: Yeah, as you specified missing response object here.

**Pull request merged**